### PR TITLE
Update documentation for pointFeature.

### PR DESCRIPTION
**Pull request by manthey**
This also touches documentation on util/clustering.js, though that could use some refactoring so that clusters are computed in map gcs rather than in feature gcs (usually interface gcs) and the cluster size is given in simpler units (such as display pixels).

This fixes a number of trivial documentation issues in other files, too.

### CODE REVIEW

File: src/gl/pointFeature.js [hunk 1]
```diff
@@ -365,13 +390,33 @@ var gl_pointFeature = function (arg) {
...
    */
   this.verticesPerFeature = function () {
     var unit = pointPolygon(0, 0, 1, 1);
     return unit.length / 2;
   };
 
+  /**
+   * Set style(s) from array(s).  For each style, the array should have one
+   * value per data item.  The values are not converted or validated.  Color
+   * values should be {@link geo.geoColorObject}s.  If invalid values are given
```
Comment (aashish24) [inline src/gl/pointFeature.js]: is "s" after the "}" intentional?

File: src/pointFeature.js [hunk 2]
```diff
@@ -2,13 +2,59 @@ var inherit = require('./inherit');
...
+ *
+ * @typedef {object} geo.pointFeature.clusteringSpec
+ * @property {number} [radius=0.05] This is combined with the `width` and
+ *   `height` to determine how close points need to be to each other to be
+ *   clustered.
+ * @property {number} [maxZoom=18] Never cluster above this zoom level.
+ * @property {number} [width=256]
+ * @property {number} [height=256]
+ */
+// TODO: refactor point clustering to (a) not be based on window size, (b) use
```
Comment (aashish24) [inline src/pointFeature.js]: perhaps we could create an issue for this and remove it from here?

File: src/gl/pointFeature.js (hunk 1 above)
Comment (manthey) [reply src/gl/pointFeature.js]: Yes.  It makes the docs read 'Color values should be `geo.geoColorObject`s.', which seemed more grammatical, but I'm not committed to the 's'.

File: src/pointFeature.js (hunk 2 above)
Comment (manthey) [reply src/pointFeature.js]: Sure.

Comment (manthey) [reply src/pointFeature.js]: Apparently I already did (see issue #853).  Do you want it removed?  It stands as a warning to anyone using point clustering.

File: src/gl/pointFeature.js (hunk 1 above)
Comment (aashish24) [reply src/gl/pointFeature.js]: I am fine with this. You could have said that 'Color' values should a a geo.geoColorObject.

File: src/pointFeature.js (hunk 2 above)
Comment (aashish24) [reply src/pointFeature.js]: I would remove it as it is not consistent and a user would not dig in the code as much I think.

File: src/gl/pointFeature.js (hunk 1 above)
Comment (manthey) [reply src/gl/pointFeature.js]: I reworded it a little.

File: src/pointFeature.js (hunk 2 above)
Comment (manthey) [reply src/pointFeature.js]: Done.

**Pull request merged**